Get started page UX improvements: add FAQs card

ianlin opened the pull request:
### Changes proposed in this Pull Request:

Part of #1392, this PR adds the FAQs card to the new get started page.

Design: eQ9O4m2flzrcAiDMXkOW0m-fi-7350%3A121282

The latest copies and the links of the FAQ can be found in the `Marketing copy document` in this post: pcTzPl-i7-p2#design-requirements. Need to request access in order to see the content.

### Screenshots:

#### Desktop

<img width="1442" alt="Screenshot 2022-05-03 at 14 45 20" src="https://user-images.githubusercontent.com/914406/166413944-b3e69541-4b4b-4618-b98b-0d088f144420.png">

<img width="1345" alt="gla-get-started-faq-desktop-expanded" src="https://user-images.githubusercontent.com/914406/166412619-adef5854-3178-469e-8e28-43d5ef6da7a9.png">


#### Mobile

<img width="859" alt="Screenshot 2022-05-03 at 14 46 11" src="https://user-images.githubusercontent.com/914406/166414005-9b37bbd0-32b6-418a-affa-a5e7be94fb75.png">

<img width="549" alt="gla-get-started-faq-mobile-expanded" src="https://user-images.githubusercontent.com/914406/166412669-d27d3763-9d88-4e58-adbb-7ff58258f5a7.png">

#### Tablet

<img width="900" alt="Screenshot 2022-05-03 at 14 46 28" src="https://user-images.githubusercontent.com/914406/166414029-f32e3317-87e1-4e20-ade7-9cf6b3b2dcfb.png">

<img width="777" alt="gla-get-started-faq-tablet-expanded" src="https://user-images.githubusercontent.com/914406/166412672-1b096b32-f778-4958-aef9-3425f0cb43e2.png">


### Detailed test instructions:

1. Go to GLA's get started page: `https://domain.test/wp-admin/admin.php?page=wc-admin&path=%2Fgoogle%2Fstart`
2. Check the sixth card of the get started page, the UI should align with the design.
3. Use the mobile view from the dev tool, the text of the longer question shouldn't be overlapped with the caret down icon like the below image:
    - <img width="561" alt="Screenshot 2022-05-03 at 14 33 14" src="https://user-images.githubusercontent.com/914406/166412912-c1e8eb06-c470-4d62-9bbd-33d4ca38f672.png">

### Changelog entry

> Update - Add the FAQs card for UX improvements on get started page.

ianlin committed:
Move get-started-page/faqs.js to get-started-page/faqs/index.js

ianlin committed:
Update faqs panel header's style in get started faqs to align design

ianlin committed:
Update the 1st FAQ in get started page

ianlin committed:
Update the 2nd FAQ in get started page

ianlin committed:
Update the 3rd FAQ in get started page

ianlin committed:
Update the 4th FAQ in get started page

ianlin committed:
Update the 5th FAQ in get started page

ianlin committed:
Update the 6th FAQ in get started page

ianlin committed:
Update the 7th FAQ in get started page

ianlin committed:
Update the 8th FAQ in get started page

ianlin committed:
Fix the FAQ questions would overlap with the caret down icon

On some mobile devices like iPhone 12 Pro.

puntope commented on js/src/components/faqs-panel/index.js:
```diff
@@ -30,11 +31,12 @@ const getPanelToggleHandler = ( trackName, id ) => ( isOpened ) => {
  * @param {Object} props React props.
  * @param {string} props.trackName The track event name to be recorded when toggling on FAQ items.
  * @param {Array<FaqItem>} props.faqItems FAQ items for rendering.
+ * @param {string} props.className The class name for this component.
```
💅  

props.className  Is optional

ianlin replied:
Thanks, updated in 43a1d02.

puntope replied:
I still see reduced space in 

`How do I sync my products to Google free listings?` and `Can I run both Shopping ads and free listings campaigns at the same time?`

![Uploading Screenshot 2022-05-04 at 11.56.06.png…]()

puntope commented:
I saw some inconsistencies between spacing.

<img width="916" alt="Screenshot 2022-05-03 at 13 12 01" src="https://user-images.githubusercontent.com/5908855/166430267-0e37ba4d-f985-4f17-bcdb-20273f44fb20.png">

<img width="980" alt="Screenshot 2022-05-03 at 13 11 57" src="https://user-images.githubusercontent.com/5908855/166430281-6064cfb1-dc22-4151-895b-e122f694fc04.png">

puntope commented:
💅  To consider a refactor of this:

<img width="480" alt="Screenshot 2022-05-03 at 13 19 47" src="https://user-images.githubusercontent.com/5908855/166430413-215f303e-3ea9-4687-b7fb-7d8b52e26933.png">

Maybe using flex styling instead of absolute positioning the SVG.

puntope commented:
> The latest copies and the links of the FAQ can be found in the Marketing copy document in this post: pcTzPl-i7-p2#design-requirements. Need to request access in order to see the content.


Requested access.

puntope reviewed (commented):
Thanks for the implementation 🙏 

Left some comments and suggestions.

ianlin committed:
Indicates className is optional in JSDoc for FaqsPanel component

ianlin committed:
Apply <p> to every row of FAQs so they have same line heights

ianlin commented:
@puntope Thanks for the review!

> Maybe using flex styling instead of absolute positioning the SVG.

Did you still see the overlapping on the text and the caret down icon? Because there was actually a simple commit 5aa5035 that might fix it by simply add more padding to the right.

Since we are using [Panel component from Gutenberg ](https://github.com/WordPress/gutenberg/blob/1326a46cf10ad54e897d1f3c9aaa0e385aba896a/packages/components/src/panel/body.js#L110-L116) for the FAQs so it's a bit tricky to override its styles using Flex styling, as they put the svg before the text in the code there. Let me know if I miss anything, thanks!

puntope commented:
> Flex styling, as they put the svg before the text in the code there

flex-direction: row-reverse; would help with that. But I'll check with your fix 👌

puntope commented:
In regards https://github.com/woocommerce/google-listings-and-ads/pull/1470/commits/f82e29a8c9b01d9dd83881f774f527caa20d7316

I still see reduced space in 

`How do I sync my products to Google free listings?` and `Can I run both Shopping ads and free listings campaigns at the same time?`

<img width="677" alt="Screenshot 2022-05-04 at 11 56 06" src="https://user-images.githubusercontent.com/5908855/166642450-fb031708-6b09-4acb-8a84-ee5fac7af26b.png">

puntope commented:
In regards https://github.com/woocommerce/google-listings-and-ads/commit/5aa5035ca3a8de1ff8e4ad1cdfd56881ef539f23

I still see the chevron quite close to the headline

<img width="579" alt="Screenshot 2022-05-04 at 12 01 11" src="https://user-images.githubusercontent.com/5908855/166643093-92a704ef-66b0-42b5-8c51-4fdf78c13092.png">

ianlin committed:
Apply <p> to every row of FAQs so they have same line heights

ianlin committed:
Use flex to wrap the gutenberg panel's svg icon

Originally it used absolute position so the longer text would be
overlapped with the icon.

ianlin committed:
Adjust FAQ's question title's font style to align the design

ianlin commented:
Hey @puntope thanks for the tip! I've updated the codes and it's ready for the review again.

> flex-direction: row-reverse; would help with that.

Awesome, it worked, thanks and TIL! Updated in 641a397. I added it in `js/src/components/faqs-panel/index.scss` so both this component and `js/src/setup-mc/setup-stepper/setup-accounts/faqs.js` can use the same change.

> I still see reduced space in
`How do I sync my products to Google free listings?` and `Can I run both Shopping ads and free listings campaigns at the same time?`

Ah thanks for the sharp eyes, finally got what you meant. It's the difference for the bottom spaces. Updated in 
1232c2e, thanks!

Finally I also add a commit 486cc21 to make the question title's font style to be aligned with the design.

puntope commented on js/src/components/faqs-panel/index.scss:
```diff
@@ -2,4 +2,19 @@
+			justify-content: space-between;
+
+			.components-panel__arrow {
+				position: unset;
+				right: unset;
+				top: unset;
+				transform: unset;
```
```
right: unset;
top: unset;
```

This two seems no necessary. Since right and top are not affecting static positioned elements.

ianlin replied:
Thanks! Updated in 9e15573.

puntope reviewed (approved):
✅  APPROVED 

- Checked the copy and is correct
- Checked the design and match the Figma

Left some pending comment regarding the unnecessary override of right, and top props. But approved in advance.

ianlin committed:
Remove unnecessary overrides for Panel component
